network/p2p/sampler.go

```diff
@@ -0,0 +1,59 @@
+package p2p
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+)
+
+type SamplingFilter func(ctx context.Context, nodeID ids.NodeID) bool
+
+// NewPeerValidatorFilter returns a filter that only samples other peer
+// validators
+func NewPeerValidatorFilter(self ids.NodeID, validators ValidatorSet) SamplingFilter {
+	return func(ctx context.Context, nodeID ids.NodeID) bool {
+		return nodeID != self && validators.Has(ctx, nodeID)
+	}
+}
```

Comment:
I think this should be 2 different filters. One for filtering a set of peers. Another for filtering for validators.